lib/store/inventory.ex

```diff
@@ -309,11 +309,69 @@ defmodule Store.Inventory do
       if order_product != nil do
         {:error, "Product already in cart"}
       else
-        create_order_product(%{order_id: order.id, product_id: product.id})
+        {val, _} = Integer.parse(product_params["quantity"])
+        size = product_params["size"]
+
+        if update_stock_sizes(product, size, val) == {:error, "Not enough stock"} do
+          {:error, "Not enough stock"}
+        else
+          create_order_product(%{
+            order_id: order.id,
+            product_id: product.id,
+            quantity: val,
+            size: size
+          })
+        end
       end
     else
       {:ok, order} = create_order(%{user_id: user.id})
-      create_order_product(%{order_id: order.id, product_id: product.id})
+      {val, _} = Integer.parse(product_params["quantity"])
+      size = product_params["size"]
+
+      if update_stock_sizes(product, size, val) == {:error, "Not enough stock"} do
+        {:error, "Not enough stock"}
+      else
+        create_order_product(%{
+          order_id: order.id,
+          product_id: product.id,
+          quantity: val,
+          size: size
+        })
+      end
```

Comment:
This code is repeated twice. You should refactor it into a function.

Reply:
The code is intentionally duplicated because each function contains its own unique errors, and I want to encapsulate them as thoroughly as possible.
Do you have any suggestions, how i should change this?

Reply:
I'm talking about the code between lines 312-324 and lines 331-340 😅